src/sentry/relay/config/__init__.py

```diff
@@ -178,7 +178,10 @@ def _get_project_config(project, full_config=True, project_keys=None):
     if allow_dynamic_sampling:
         dynamic_sampling = project.get_option("sentry:dynamic_sampling")
         if dynamic_sampling is not None:
-            cfg["config"]["dynamicSampling"] = dynamic_sampling
+            # filter out rules that do not have active set to True
+            cfg["config"]["dynamicSampling"] = {
+                "rules": [r for r in dynamic_sampling["rules"] if r.get("active")]
```

Comment:
It is not available to public, there should be no "legacy" rules

Reply:
This is a valid concern. consider what absence of "active" means. with this change, all rules will be disabled upon deploy. that might be fine

Reply:
If we really don't care about pre-existing rules (not even for some test projects that we set up for development / dogfooding) then we're good. I would just make sure that the UI also displays those legacy rules as disabled.

Reply:
@ahmedetefy may i suggest a `disabled` flag? it's easier to handle, its default value `None/False` is the one you want as well.

Reply:
What do you mean by `disabled` flag? As in rename "active" ? The current UI calls it `active`, why would we rename?

Reply:
> I would just make sure that the UI also displays those legacy rules as disabled.

@priscilawebdev can you confirm this?

Reply:
> just a random idea to change active=True to disabled=False, then you can keep treating None and False the same

Honestly, I don't know if this is necessary, the UI calls it active + its not in prod + from looking at both sentry, and javascript we are not dogfooding either. 
Also for new rules, I set a default to `False` on the serializer -> https://github.com/getsentry/sentry/pull/36416

Can't really imagine a situation where this would matter